critwall/src/main/java/org/opencadc/critwall/BucketSelector.java

```diff
@@ -67,18 +67,83 @@
 
 package org.opencadc.critwall;
 
+import ca.nrc.cadc.util.StringUtil;
+import java.util.Iterator;
+import java.util.TreeSet;
 import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+
 
-/**
- *
- * @author pdowler
- */
 public class BucketSelector {
-    private final Logger log = Logger.getLogger(BucketSelector.class);
-    public final String bucketSelector;
+    private static final Logger log = Logger.getLogger(BucketSelector.class);
+    // range as entered in .properties file
+    private final String bucketSelectors;
+    // parsed values
+    private final String rangeMin;
+    private final String rangeMax;
+    private static final String HEXVALUES = "0123456789abcdef";
+    // generated list of bucket selectors
+    private TreeSet<String> bucketList = new TreeSet<String>();
+
+    public Iterator<String> getBucketIterator() {
+        return bucketList.iterator();
+    }
+
+    public BucketSelector(String selectors) {
+        InventoryUtil.assertNotNull(BucketSelector.class, "selectors", selectors);
+        this.bucketSelectors = selectors;
+
+        String[] minMax = this.bucketSelectors.split("-");
```

Comment:
line 97: StringBuffer is thread safe and performs poorly; always use StringBuilder unless you need thread safety (same API)

line 106 and 120: why not minMax[i].trim()? don't apply toLowerCase - values in config should be lower case or fail the subsequent validation; the storage dm specifies what the values look like and upper and lower case are *not* equivalent

line 108 and 122: these check for a token that was trimmed to 0-length? but the lines 111 and 125 look for those empty strings in the alphabet anyway... the error message will be "invalid hex value: \n" (confusing)

Layerrs of nested if/else are really hard to follow. We used to have a code-style check that counted the number of different paths through a method (~complexity) and nested if/else is remarkably good at creating excess complexity. More straightforward would be to:
```
rangeMin = minMax[0].trim();
rangeMax = rangeMin;
if (minMax.length == 2) {
   rangeMax = minMax[1].trim();
}
// validate length of tokens

// validate characters in tokens

// validate order

// construct set
```

where each section completes or throws an exception (which also reduces nested if/else - for example, the else on line 104 can be removed because minMax.length has to be 1 or 2 for all subsequent code)